homeassistant/components/mazda/switch.py

```diff
@@ -0,0 +1,73 @@
+"""Platform for Mazda switch integration."""
+from pymazda import Client as MazdaAPIClient
+
+from homeassistant.components.switch import SwitchEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from . import MazdaEntity
+from .const import DATA_CLIENT, DATA_COORDINATOR, DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up the switch platform."""
+    client = hass.data[DOMAIN][config_entry.entry_id][DATA_CLIENT]
+    coordinator = hass.data[DOMAIN][config_entry.entry_id][DATA_COORDINATOR]
+
+    async_add_entities(
+        MazdaChargingSwitch(client, coordinator, index)
+        for index, data in enumerate(coordinator.data)
+        if data["isElectric"]
+    )
+
+
+class MazdaChargingSwitch(MazdaEntity, SwitchEntity):
+    """Class for the charging switch."""
+
+    def __init__(
+        self,
+        client: MazdaAPIClient,
+        coordinator: DataUpdateCoordinator,
+        index: int,
+    ) -> None:
+        """Initialize Mazda charging switch."""
+        super().__init__(client, coordinator, index)
+
+        self._attr_name = f"{self.vehicle_name} Charging"
+        self._attr_unique_id = self.vin
+        self._attr_icon = "mdi:ev-station"
```

Comment:
This is static. You can make it a class attribute

Reply:
Nice catch. Fixed